homeassistant/components/mqtt/client.py

```diff
@@ -247,7 +261,7 @@ class Subscription:
 
     topic: str = attr.ib()
     matcher: Any = attr.ib()
-    job: HassJob[[ReceiveMessage], Coroutine[Any, Any, None] | None] = attr.ib()
```

Comment:
@epenet I want to change the type here to just HassJob. Without the change mypy is complaining about the first argument of `hass.async_run_hass_job(`
https://github.com/home-assistant/core/pull/77972/files#diff-6104fc3fa32da922a306affdf3ac8882f9e4d2f89d78970724e0f164d5b1fd6cR654

Reply:
Maybe @cdce8p has ideas about this one. If not, I'll try to take a look - probably not before Monday though

Reply:
> I want to change the type here to just HassJob. Without the change mypy is complaining about the first argument of `hass.async_run_hass_job(` https://github.com/home-assistant/core/pull/77972/files#diff-6104fc3fa32da922a306affdf3ac8882f9e4d2f89d78970724e0f164d5b1fd6cR654
> 
> Can I just do that?

That is an issue with mypy. The type itself is correct which can be checked with pyright. I would suggest adding `type: ignore[arg-type]` to the line you linked above.